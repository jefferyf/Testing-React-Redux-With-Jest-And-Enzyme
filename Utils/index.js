import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from './../src/reducers/index';
import { middlewares } from './../src/createStore';

export const checkProps = (component, expectedProps) => {
  const propsError =  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsError;
};

export const findByTestAttr = (wrapper, attr) => {
  return wrapper.find(`[data-test='${attr}']`);
};

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initialState);
}