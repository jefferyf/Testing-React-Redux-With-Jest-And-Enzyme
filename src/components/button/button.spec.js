import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr, checkProps } from '../../../Utils';
import SharedButton from './button';

describe('SharedButton Component', () => {
  describe('Checking PropTypes', () => {
    it('should not throw a warning', () => {
      const expectedProps = {
        buttonText: 'Button Text',
        emitEvent: () => {

        }
      };
      const propsError = checkProps(SharedButton, expectedProps);
      expect(propsError).toBeUndefined();
    });
  });

  describe('Snapshot Tests', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<SharedButton />);
      wrapper.setProps({ buttonText: 'Press Me'});
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Renders', () => {
    let wrapper,
        props;

    beforeEach(() => {
      props = {
        buttonText: 'Button Text',
        emitEvent: jest.fn()
      };
      wrapper = shallow(<SharedButton {...props} />);
    });

    it('Should render a button', () => {
      expect(findByTestAttr(wrapper, 'buttonComponent').length).toBe(1);
    });

    it('should call the emit event when clicked', () => {
      findByTestAttr(wrapper, 'buttonComponent').simulate('click');
      expect(props.emitEvent).toHaveBeenCalled();
    });
  });
});