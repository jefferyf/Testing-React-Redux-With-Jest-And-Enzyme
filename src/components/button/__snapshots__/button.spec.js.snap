// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedButton Component Snapshot Tests renders correctly 1`] = `
<button
  data-test="buttonComponent"
  onClick={[Function]}
>
  Press Me
</button>
`;
